File: PAT/basic_level/1023_组个最小数/min_permutation.c
#include <stdio.h>

int main(){
	int i;
    int numList[16];

	for(i=0; i<10; i++){
		scanf("%d", &numList[i]);
	}

	for(i=1; i<10; i++){
		if( 0!=numList[i] ){
			printf("%d", i);
            numList[i]--;
			break;
		}
	}

	for(i=0; i<10; i++){
		while( 0!=numList[i] ){
			printf("%d", i);
			numList[i]--;
		}
	}

	return 0;
}

File: PAT/basic_level/1023_组个最小数/min_permutation.py
from __future__ import print_function

numList = raw_input().split()
numList = map(int, numList)

for i in range(1, 10):
    if 0!=numList[i]:
        print(i, end='')
        numList[i] -= 1
        break

for i in range(10):
    while numList[i]:
        print(i, end='')
        numList[i] -= 1
